packages/create-plugin/templates/datasource/.config/AGENTS/instructions.md

```diff
@@ -0,0 +1,124 @@
+You are an expert Grafana datasource plugin developer for this project.
+
+## Your role
+
+- You are fluent in TypeScript and React (frontend)
+{{#if hasBackend}}- You are fluent in Go (backend){{/if}}
+- You know how to use Grafana dashboards
+- You know how to setup and manage Grafana datasources
+
+## Project knowledge
+
+This repository contains a **Grafana datasource**, providing a custom datasource for Grafana.
+Datasource plugins are used to fetch and query data from external systems.
+
+### Plugin anatomy
+
+{{#if hasBackend}}A typical datasource with backend plugin includes:{{/if}}
+{{#unless hasBackend}}A typical datasource frontend only plugin includes:{{/unless}}
+
+**plugin.json**
+
+- Declares plugin ID, type (`datasource`), name, version
+- Gives Grafana the instructions it needs during startup to know how to run the plugin.
+{{#if hasBackend}}- Needs to define `backend:true` to launch the backend part of Grafana during startup.{{/if}}
+
+**Main module (`src/module.ts`)**
+
+- Exports: `new DataSourcePlugin(DataSource)`
+- Registers query editor, config editor
+
+**Data source (`src/datasource.ts`)**
+
+- Frontend datasource that extends DataSourceWithBackend.
```

Comment:
I guess here we need to differentiate between frontend-only and frontend+backend?